homeassistant/components/solarlog/coordinator.py

```diff
@@ -35,6 +36,7 @@ def __init__(self, hass: HomeAssistant, entry: SolarlogConfigEntry) -> None:
         )
 
         host_entry = entry.data[CONF_HOST]
+        password = entry.data.get("password", "")
```

Comment:
Use constants, why do we set it to `""` as default?

Reply:
Because we cannot pass None as argument.
If user does not check "has_password", property will not be set in config flow, so we cannot be sure this exists.

Reply:
No I mean, why not 
```suggestion
        password = entry.data.get("password")
```

Reply:
But _why_ must it be `str` and not `None`?

Reply:
There is no specific reason besides that it seemed simpler and more consistent to me to have only a str in the library.

Reply:
I think it's more consistent to make it nullable. And otherwise we can make it None here and pass it to the library as `password or ""`, just to make sure we keep the code to do what it expects

Reply:
Is there a reason that speaks against get? Instead of making it None I would then rather set it to "" in config flow and migration...

Reply:
Like I'd rather not have it than set to something fake, how do you think about that?